### PR TITLE
Use 'global_pose' terminology for submaps.

### PR DESCRIPTION
**Pull request by wohe**
This follows #624.

### CODE REVIEW

File: cartographer/mapping_2d/sparse_pose_graph.h [hunk 1]
```diff
@@ -65,11 +65,11 @@ class SparsePoseGraph : public mapping::SparsePoseGraph {
...
-  // Adds a new node with 'constant_data' and a 'pose' that will later be
-  // optimized. The 'pose' was determined by scan matching against
-  // 'insertion_submaps.front()' and the scan was inserted into the
-  // 'insertion_submaps'. If 'insertion_submaps.front().finished()' is
-  // 'true', this submap was inserted into for the last time.
+  // Adds a new node with 'constant_data'. Its 'constant_data->local_pose' was
+  // determined by scan matching against 'insertion_submaps.front()' and the
+  // scan was inserted into the 'insertion_submaps'. If
+  // 'insertion_submaps.front().finished()' is 'true', this submap was inserted
+  // into for the last time.
```
Comment (gaschler) [inline cartographer/mapping_2d/sparse_pose_graph.h]: Word missing "into for"?

File: cartographer/mapping_3d/sparse_pose_graph.h [hunk 2]
```diff
@@ -65,11 +65,11 @@ class SparsePoseGraph : public mapping::SparsePoseGraph {
...
-  // Adds a new node with 'constant_data' and a 'pose' that will later be
-  // optimized. The 'pose' was determined by scan matching against
-  // 'insertion_submaps.front()' and the scan was inserted into the
-  // 'insertion_submaps'. If 'insertion_submaps.front().finished()' is
-  // 'true', this submap was inserted into for the last time.
+  // Adds a new node with 'constant_data'. Its 'constant_data->local_pose' was
+  // determined by scan matching against 'insertion_submaps.front()' and the
+  // scan was inserted into the 'insertion_submaps'. If
+  // 'insertion_submaps.front().finished()' is 'true', this submap was inserted
+  // into for the last time.
```
Comment (gaschler) [inline cartographer/mapping_3d/sparse_pose_graph.h]: Same here.

File: cartographer/mapping_2d/sparse_pose_graph.h (hunk 1 above)
Comment (wohe) [reply cartographer/mapping_2d/sparse_pose_graph.h]: Not really, it's passive voice. "For the last time" data was inserted into the submap. I could change it to
```
If 'insertion_submaps.front().finished()' is 'true', data was inserted into this submap for the last time.
```
if this is easier to parse?

Comment (wohe) [reply cartographer/mapping_2d/sparse_pose_graph.h]: Changed it.

Comment (gaschler) [reply cartographer/mapping_2d/sparse_pose_graph.h]: Thanks, this I can parse better.

**Pull request merged**